### PR TITLE
feat(crashlytics): add automatic Crashlytics symbol uploads for iOS & macOS apps

### PR DESCRIPTION
**Pull request by elenadoty**
## Description

Adds a script that automatically adds Crashlytics's upload-symbols tool to the run script of Xcode projects for iOS apps when the Crashlytics plugin is added as a dependency to the project.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x]  I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_crashlytics/firebase_crashlytics/ios/crashlytics_add_upload_symbols [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+puts(project_path)
+unless (File.exist?(project_path)) 
+    abort("Project at #{project_path} does not exist. Please incorporate Crashlytics upload symbols manually.\n");
+end
+
+# Default to Flutter arguments if no other upload-symbols inputs are provided.
+if(ARGV.length == 3)
+    upload_symbols_args = ARGV[2]
+else
+    upload_symbols_args = "--flutter-project \"$PROJECT_DIR/firebase_app_id_file.txt"
```
Comment (Salakar) [inline packages/firebase_crashlytics/firebase_crashlytics/ios/crashlytics_add_upload_symbols:20]: Is this .txt file in relation to the CLI PR https://github.com/invertase/flutterfire_cli/pull/43?

Comment (elenadoty) [reply packages/firebase_crashlytics/firebase_crashlytics/ios/crashlytics_add_upload_symbols]: Yes - I changed the file format to be more structured (now is json) so need to update this!

**Comment by Salakar** [comment]
Thanks again for this 🎉  I will merge this now for todays release. The e2e job failures are relating to an issue with the Firebase Storage emulator - we're working on this excluding these (separate PR) for now until emulator issue can be resolved. iOS and macOS are building fine with script running successfully.

**Comment by cuong0993** [comment]
Appreciate your work but this is a breaking change. It forces projects to use flutterfire_cli to generate firebase_app_id_file.json file. 

I do not use flutterfire_cli. I prefer to download GoogleService-Info.plist like I did in native.

Can we change the implementation to read googleAppId from an environment variable or something does not depend on flutterfire_cli?

**Pull request merged**